### DISCUSSION TITLE
Can not override log.transports.file.archiveLog

### DISCUSSION THREAD

**Discussion by gorn** [megahertz/electron-log#292] [Q&A] [answered]
This might be something very basic, but I am desperate - I try to keep several logs by overriding the archiveLog callback like this:

````typescript 
import log from "electron-log"
import path from "path"
import fs from "fs"
import moment from "moment"

log.transports.file.archiveLog = (file: string) => {
  const fileStr = file.toString()
  const fileInfo = path.parse(fileStr)
  try {
    fs.renameSync(fileStr, path.join(fileInfo.dir, fileInfo.name + "." + moment().format('YYYYMMDDHHmmss') + fileInfo.ext))
  } catch (e) {
    console.warn("Could not rotate log", e)
  }
}
````

but it does not help. Something changes, because the file is not rotated at all and new messages are added to it only, but it is not what I need. I also tried debug it like this, but I get no message in the console

````typescript 
log.transports.file.archiveLog = (file: string) => {
  console.log('LOGGING')
}
````

**Comment by megahertz** [comment, marked as answer]
Firstly, you need to make sure that you code is executed. Try to set `maxSize` to 1 for debugging your custom handler.

**Reply by gorn** [reply]
That was it, I have misunderstood the philosophy of this setting. Thanks (namely is seemed to me that log is rotated everytime plus when it reaches maxSize)
